demo/src/plot_demo.rs

```diff
@@ -1101,6 +1107,61 @@ impl ChartsDemo {
     }
 }
 
+#[derive(PartialEq, serde::Deserialize, serde::Serialize, Default)]
+struct UserdataDemo {}
+
+struct DemoPoint {
+    x: f64,
+    y: f64,
+    custom_thing: bool,
```

Comment:
This makes it sounds like some demo points has a custom thing, other don't. I think this feature deserves a more realistic example